From 23b2900893afaca407c111c27d970b4818d94391 Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Thu, 22 Dec 2016 16:52:25 +0200
Subject: [PATCH 0682/1345] mmc: dts: Apply changes according to xenon driver
 v4 to DTS

commit  10786481b0704c01899e18bdb9884a551eac7ee3 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

Update DTS files of platfoms using xenon MMC unit according
to changes introduced by the mainline driver v4 backport.

Change-Id: If104f2eb59e33a506b68410223ad42282302d1b1
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/34942
Reviewed-by: Omri Itach <omrii@marvell.com>
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/marvell/armada-3720-community.dts     |    1 +
 .../boot/dts/marvell/armada-37x0-customer.dts      |    5 +++--
 arch/arm64/boot/dts/marvell/armada-7040-db.dtsi    |    3 +++
 arch/arm64/boot/dts/marvell/armada-8040-db.dtsi    |    4 ++++
 arch/arm64/boot/dts/marvell/armada-cp110.dtsi      |    3 +--
 5 files changed, 12 insertions(+), 4 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-3720-community.dts b/arch/arm64/boot/dts/marvell/armada-3720-community.dts
index a3f7099..33f92ab 100644
--- a/arch/arm64/boot/dts/marvell/armada-3720-community.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3720-community.dts
@@ -116,6 +116,7 @@
 				bus-width = <4>;
 				cd-gpios = <&gpio_nb 3 GPIO_ACTIVE_LOW>;
 				no-1-8-v;
+				marvell,pad-type = "sd";
 				status = "okay";
 
 				pinctrl-names = "default";
diff --git a/arch/arm64/boot/dts/marvell/armada-37x0-customer.dts b/arch/arm64/boot/dts/marvell/armada-37x0-customer.dts
index f0f6600..0fbfd95 100644
--- a/arch/arm64/boot/dts/marvell/armada-37x0-customer.dts
+++ b/arch/arm64/boot/dts/marvell/armada-37x0-customer.dts
@@ -197,6 +197,7 @@
 				bus-width = <4>;
 				cd-gpios = <&gpio_sb 2 GPIO_ACTIVE_LOW>;
 				vqmmc-supply = <&vcc_sd_reg1>;
+				marvell,pad-type = "sd";
 				status = "disabled";
 				/* UPDATE */
 				pinctrl-names = "default";
@@ -204,10 +205,10 @@
 			};
 
 			sdhci0: sdhci@d8000 {
-				xenon,fixed-1-8v-pad-ctrl;
+				marvell,pad-type = "fixed-1-8v";
+				marvell,xenon-tun-count = <9>;
 				non-removable;
 				bus-width = <8>;
-				tuning-count = <9>;
 				status = "disabled";
 				/* UPDATE */
 				pinctrl-names = "default";
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi b/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
index 97fa0db..deb5b3b 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
@@ -109,7 +109,10 @@
 				usb-phy = <&usb3h1_phy>;
 			};
 			sdhci@780000 {
+				bus-width = <4>;
+				no-1-8-v;
 				non-removable;
+				marvell,pad-type = "sd";
 			};
 			spi@700680 {
 				spi-flash@0 {
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi b/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
index 86c4689..2f238a7 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
@@ -59,6 +59,7 @@
 	ap806 {
 		config-space@f0000000 {
 			sdhci@6e0000 {
+				marvell,xenon-emmc;
 				bus-width = <8>;
 				non-removable;
 			};
@@ -116,6 +117,9 @@
 			};
 			sdhci@780000 {
 				bus-width = <4>;
+				no-1-8-v;
+				non-removable;
+				marvell,pad-type = "sd";
 				non-removable;
 			};
 		};
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
index abef02d..bead961 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
@@ -298,12 +298,11 @@ mdio@15b000 {
 };
 
 sdhci@780000 {
-	compatible = "marvell,xenon-sdhci";
+	compatible = "marvell,armada8k-sdhci";
 	reg = <0x780000 0x300>;
 	interrupts = <ICU_GRP_NSR 27 IRQ_TYPE_LEVEL_HIGH>;
 	clock-names = "core";
 	clocks = <&cpm_syscon0 1 4>;
-	xenon,slotno = <0>;
 	dma-coherent;
 	status = "disabled";
 };
-- 
1.7.9.5

